httpd/handler.go

```diff
@@ -286,13 +286,17 @@ func (h *Handler) serveMetastore(w http.ResponseWriter, r *http.Request) {
 
 // servePing returns a simple response to let the client know the server is running.
 func (h *Handler) servePing(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusNoContent)
+	if h.server.Ready() {
+		w.WriteHeader(http.StatusNoContent)
+	} else {
+		w.WriteHeader(http.StatusServiceUnavailable)
+	}
 }
 
 // serveDataNodes returns a list of all data nodes in the cluster.
 func (h *Handler) serveDataNodes(w http.ResponseWriter, r *http.Request) {
 	// Generate a list of objects for encoding to the API.
-	a := make([]*dataNodeJSON, 0)
+	var a = make([]*dataNodeJSON, 0)
```

Comment:
Can you change this back? The convention in the codebase so far has been to use `:=` unless a variable is being initialized to its zero value.